### DISCUSSION TITLE
Lazy fallback for get_prompt

### DISCUSSION THREAD

**Discussion by ogunoz** [langfuse/langfuse#2745] [Ideas]
### Describe the feature or potential improvement

At the moment, we do not prefer to use fallback argument of get_prompt method in Python SDK because we would like to fetch our fallback prompts from a DB and we would like that happen only when there is a failure. It would be awesome if there is a way to provide fallback prompt in a lazy way such as a function / lambda

Your function would look like or can be separated such as fallback_factory under another variable
```python
def get_prompt(
        self,
        name: str,
        version: Optional[int] = None,
        *,
        label: Optional[str] = None,
        type: Literal["chat", "text"] = "text",
        cache_ttl_seconds: Optional[int] = None,
        fallback: Union[Optional[List[ChatMessageDict]], Optional[str], Optional[Callable[[str, Optional[int], Optional[str]], Union[ChatMessageDict, str]]] = None,
        max_retries: Optional[int] = None,
    ) -> PromptClient:
```

Usage would be like

```python
def get_fallback_prompt(name: str, version: int | None, label: str | None) -> list[ChatMessageDict] | str:
  # Our own fallback function e.g., read from DB, read from file and return either list of chat message dict or just a string
  return ...

langfuse = Langfuse()
langfuse.get_prompt(
        name=name,
        version=version,
        label=label,
        cache_ttl_seconds=cache_tll_seconds,
        fallback=get_fallback_prompt,
)
```

### Additional information

The feature request is for Python SDK

**Comment by marcklingen** [comment]
thanks for sharing! are you interested in contributing this?

based on a change to the caching behavior, maybe this solution solves for you availability/latency goals without having to implement fallbacks:
- Changelog: https://langfuse.com/changelog/2024-09-04-prompt-management-zero-latency
- Optional: pre-fetch prompts: https://langfuse.com/docs/prompts/get-started#optional-pre-fetch-prompts-on-application-start
